apptentive/src/com/apptentive/android/sdk/module/engagement/interaction/view/NavigateToLinkInteractionView.java

```diff
@@ -17,26 +19,39 @@
  *
  * @author Sky Kelsey
  */
-public class NavigateToUrlInteractionView extends InteractionView<NavigateToLinkInteraction> {
+public class NavigateToLinkInteractionView extends InteractionView<NavigateToLinkInteraction> {
 
-	public NavigateToUrlInteractionView(NavigateToLinkInteraction interaction) {
+	public NavigateToLinkInteractionView(NavigateToLinkInteraction interaction) {
 		super(interaction);
-
 	}
 
 	@Override
 	public void show(Activity activity) {
 		super.show(activity);
 
+		boolean success = false;
 		try {
 			String url = interaction.getUrl();
 			Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse(url));
 
-			// Do this if the customer wants to launch the new URL outside the app as a separate app.
-//			intent.setFlags(Intent.FLAG_ACTIVITY_NO_HISTORY | Intent.FLAG_ACTIVITY_CLEAR_WHEN_TASK_RESET | Intent.FLAG_ACTIVITY_NEW_TASK);
+			switch (interaction.getTarget()) {
+				case New:
+					intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+					break;
+				case Self:
+					// Nothing
+					break;
+				default:
+					break;
+			}
 
 			activity.startActivity(intent);
+			success = true;
+		} catch (ActivityNotFoundException e) {
+			Log.w("NavigateToLink Error: ", e);
 		} finally {
+			String data = String.format("{\"url\":\"%s\",\"target\":\"%s\",\"success\":%b}", interaction.getUrl(), interaction.getTarget().lowercaseName(), success);
```

Comment:
do you worry about anything potentially being in the url that needs escaping before going into JSON?